crates/circuit/src/classical/expr/stretch.rs

```diff
@@ -0,0 +1,139 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use crate::classical::expr::{ExprKind, PyExpr};
+use crate::classical::types::Type;
+use crate::imports::UUID;
+use pyo3::prelude::*;
+use pyo3::types::{IntoPyDict, PyTuple};
+use pyo3::{intern, IntoPyObjectExt};
+use uuid::Uuid;
+
+/// A stretch variable.
+#[derive(Clone, Debug, PartialEq, Hash)]
+pub struct Stretch {
+    pub uuid: u128,
+    pub name: String,
+}
+
+impl<'py> IntoPyObject<'py> for Stretch {
+    type Target = PyAny;
+    type Output = Bound<'py, PyAny>;
+    type Error = PyErr;
+
+    fn into_pyobject(self, py: Python<'py>) -> Result<Self::Output, Self::Error> {
+        Ok(Bound::new(py, (PyStretch(self), PyExpr(ExprKind::Stretch)))?.into_any())
+    }
+}
+
+impl<'py> FromPyObject<'py> for Stretch {
+    fn extract_bound(ob: &Bound<'py, PyAny>) -> PyResult<Self> {
+        let PyStretch(s) = ob.extract()?;
+        Ok(s)
+    }
+}
+
+/// A stretch variable.
+///
+/// In general, construction of stretch variables for use in programs should use :meth:`Stretch.new`
+/// or :meth:`.QuantumCircuit.add_stretch`
+#[pyclass(eq, hash, frozen, extends = PyExpr, name = "Stretch", module = "qiskit._accelerate.circuit.classical.expr")]
+#[derive(PartialEq, Clone, Debug, Hash)]
+pub struct PyStretch(Stretch);
+
+#[pymethods]
+impl PyStretch {
+    #[new]
+    fn new0(var: &Bound<PyAny>, name: String) -> PyResult<Py<Self>> {
```

Comment:
As mentioned in [this comment](https://github.com/Qiskit/qiskit/pull/14068/files#r2045333476)
```suggestion
    fn py_new(var: &Bound<PyAny>, name: String) -> PyResult<Py<Self>> {
```